command/operator_diagnose_test.go

```diff
@@ -237,10 +169,20 @@ func TestOperatorDiagnoseCommand_Run(t *testing.T) {
 				cmd.Run(tc.args)
 				result := cmd.diagnose.Finalize(context.Background())
 
-				for i, exp := range tc.expected {
-					act := result.Children[i]
-					if err := compareResult(t, exp, act); err != nil {
-						t.Fatalf("%v", err)
+				for _, exp := range tc.expected {
+					found := false
+					// Check them all so we don't have to be order specific
+					for _, act := range result.Children {
+						if exp.Name == act.Name {
```

Comment:
I think this won't account for improperly nested checks. For example, if a stanza exists as a child of another stanza and is meant instead to be a top level stanza, this test will still pass because of the nesting here.

Reply:
Ooh, good point.